plugins/woocommerce-beta-tester/src/product-editor-dev-tools/user-preferences-panel.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { useUserPreferences, UserPreferences } from '@woocommerce/data';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { TabPanel } from './tab-panel';
+
+declare global {
+	interface Window {
+		__wcbt: object;
+	}
+}
+
+type UserPreferenceProp = keyof UserPreferences;
+
+export function UserPreferencesTabPanel( {
+	isSelected,
+}: {
+	isSelected: boolean;
+} ) {
+	const { updateUserPreferences, ...userPreferences } = useUserPreferences();
+
+	const update = useCallback(
+		(
+			preferences: UserPreferences | UserPreferenceProp,
+			value?,
+			force = false
+		) => {
+			const dataToUpdate =
+				typeof preferences === 'string'
+					? { [ preferences ]: value }
+					: preferences;
+
+			/*
+			 * When force is not True,
+			 * only update the preferences already defined
+			 * @todo: consider to implement straight in the data layer.
+			 */
+			if ( ! force ) {
+				Object.keys( dataToUpdate ).forEach( ( key: string ) => {
+					const userPreferenceKey = key as UserPreferenceProp;
+					if (
+						! userPreferences.hasOwnProperty( userPreferenceKey )
+					) {
+						delete dataToUpdate[ userPreferenceKey ];
+					}
+				} );
+			}
+
+			updateUserPreferences( dataToUpdate );
+		},
+		[ updateUserPreferences, userPreferences ]
+	);
+
+	// Expose updateUserPreferences globaly for debugging purposes.
+	window.__wcbt = {
```

Comment:
This global continues to exist even after the dev tools are closed. Not sure if that was intended or not. It doesn't seem to cause any issues (still works), so I think it is okay.

Reply:
Great catch! Although it's not a major problem, it's not exactly ideal either